packages/react-select/src/Select.js

```diff
@@ -1206,18 +1206,25 @@ export default class Select extends Component<Props, State> {
         this.selectOption(focusedOption);
         break;
       case 'Enter':
+        const targetIsButton = event.target && event.target.type === 'button';
+
+        if (targetIsButton) {
+          return;
+        }
+
         if (event.keyCode === 229) {
           // ignore the keydown event from an Input Method Editor(IME)
           // ref. https://www.w3.org/TR/uievents/#determine-keydown-keyup-keyCode
           break;
         }
+
         if (menuIsOpen) {
-          if (!focusedOption) return;
+          if (!focusedOption) break;
           if (this.isComposing) return;
           this.selectOption(focusedOption);
           break;
         }
-        return;
+        break;
```

Comment:
Why change `return` to `break` here? This seems like a breaking change should a user decide to use react-select with a `textarea`.